ui/src/app/pages/signed-in/component.ts

```diff
@@ -128,9 +177,20 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
   private navigateSignOut(): void {
     // Force a hard browser reload here. We want to ensure that no local state
     // is persisting across user sessions, as this can lead to subtle bugs.
-    window.location.assign('/');
+    window.location.assign('https://accounts.google.com/logout');
   }
 
+  closeInactivityModal(): void {
+    this.showInactivityModal = false;
+  }
+
+  get inactivityModalText(): string {
+    const timeText = environment.inactivityWarningInSecondsBefore % 60 == 0 && environment.inactivityWarningInSecondsBefore > 60 ?
+      `${environment.inactivityWarningInSecondsBefore / 60} minutes` :
+      `${environment.inactivityWarningInSecondsBefore} seconds`;
+
+    return `You've been idle for sometime. You will be logged out in ${timeText} if no activity is detected.`;
```

Comment:
I'm used to the UX here being clicking an "I'm here" button on the modal. I think this is preferred, otherwise the modal might flash in and disappear as soon as the user moves their mouse, which would be frustrating.